tiled/server/router.py

```diff
@@ -531,3 +534,60 @@ def node_full(
             )
     except UnsupportedMediaTypes as err:
         raise HTTPException(status_code=406, detail=err.args[0])
+
+
+@router.post("/node/metadata/{path:path}", response_model=schemas.PostMetadataResponse)
+def post_metadata(
+    request: Request,
+    body: schemas.PostMetadataRequest,
+    entry=Security(entry, scopes=["write:data", "write:metadata"]),
+):
+    if body.structure_family == StructureFamily.dataframe:
+        # Decode meta for pydantic validation
+        body.structure.micro.meta = deserialize_arrow(
+            base64.b64decode(body.structure.micro.meta)
+        )
+
+    try:
```

Comment:
I think this is too broad. If a node _tries_ to support writing but then encounters some error on the way, we'll tell the user "This node is not writable," when the actual problem is that writing should work but failed for some reason.

Perhaps

```py
if hasattr(entry, "post_metadata"):
    ...
else:
    raise ...
```

instead.

Reply:
What is preferred between checking `hasattr` and catching `AttributeError`?

Reply:
"Look before you leap" (`hasattr`) is preferred unless the case where it's missing is rare/exceptional. Catching exceptions is comparatively expensive and generally not the right way to do interface checking.

Another option, which has grown in popularity is to make a Protocol and then do `isinstance(entry, Writable)`. I don't think we need to get into that yet, but maybe later.